torch/csrc/jit/python/python_sugared_value.cpp

```diff
@@ -217,6 +217,54 @@ std::shared_ptr<SugaredValue> PythonModuleValue::attr(
   return toSugaredValue(member, m, loc, /*is_constant=*/true);
 }
 
+ModuleValue::ModuleValue(
+    Value* self,
+    std::shared_ptr<ConcreteModuleType> concreteType)
+    : self_(self), concreteType_(std::move(concreteType)) {
+  auto containedTypeHint = concreteType_->getContainedTypeHint();
+
+  if (containedTypeHint) {
+    // For now, only dict type containedTypeHints are supported. Generate and
+    // emit the dictionary representing the ModuleDict into the graph once so
+    // that it can be reused across lookups.
+    DictTypePtr dict_type = containedTypeHint->expect<DictType>();
+    auto* graph = self_->owningGraph();
+    std::vector<Value*> keys, values;
+
+    // Gather submodule names.
+    std::vector<std::string> submodule_names;
+    const auto& self_type = concreteType_->getJitType()->expect<ClassType>();
+    for (size_t i = 0; i < self_type->numAttributes(); ++i) {
+      const auto& attr_type = self_type->getAttribute(i);
+      if (attr_type->is_module()) {
+        if (!attr_type->isSubtypeOf(dict_type->getValueType())) {
+          auto loc = self->node()->sourceRange();
+          throw ErrorReport(loc)
+              << "Attribute " << self_type->getAttributeName(i)
+              << " is not of annotated type "
+              << dict_type->getValueType()->annotation_str();
```

Comment:
Would it be worthwhile to print more detailed reasons using the 2nd argument of `isSubtypeOf`? I don't actually know what's in the message but if it can tell me which method/argument does not match it would be very helpful.